<a href="https://colab.research.google.com/github/bidhan7/Data-Processing-Visualization-/blob/main/Big_Query_SQL.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

## Bidhan Subedi
## Data Visiualization



## SQL with BigQuery
The first thing, most of us already have an account at Cloud Resource Manager if you already have gmail logged in your computer. I required to do each of the following steps successfully in getting this document to work.  

## Before you begin


1.   Use the [Cloud Resource Manager](https://console.cloud.google.com/cloud-resource-manager) to Create a Cloud Platform project if you do not already have one.
2.   [Enable billing](https://support.google.com/cloud/answer/6293499#enable-billing) for the project.
3.   [Enable BigQuery](https://console.cloud.google.com/flows/enableapi?apiid=bigquery) APIs for the project.

In [6]:
from google.colab import auth
auth.authenticate_user()
print('Authenticated')

Authenticated


Now that I am authenticated, I can start to playing around with the dataset.  I am going to look at the bikers trip data from Austin. Yes, I have a project called `data-visualization-338723` in my BigQuery interface.

## We will use SQL to compute from dataset

`SELECT` picks the columns and `FROM` picks the table you need. 

### Computing the no of entries in the dataset.

In [7]:
%%bigquery --project data-visualization-338723
SELECT 
  COUNT(*) as total_entries
FROM `bigquery-public-data.austin_bikeshare.bikeshare_trips`

total_entries
0        1424786

That's is a lot of entries.

As there is alot of entries in the data set excel will have hard time computing. SQL will do the data manipulations on the database server much better.

 

### Computing The Longest Trip

In [11]:
%%bigquery --project data-visualization-338723
SELECT 
  MAX(duration_minutes) as Longest_trip_in_minutes
FROM `bigquery-public-data.austin_bikeshare.bikeshare_trips`

Longest_trip_in_minutes
0                    34238

The longest trip was 34238 minutes!

### Computing the average time for a trip

In [15]:
%%bigquery --project data-visualization-338723
SELECT 
  start_station_name, 
  end_station_name,
  AVG(duration_minutes) as Average_time_for_trip
FROM `bigquery-public-data.austin_bikeshare.bikeshare_trips`
WHERE start_station_name is not null
GROUP BY start_station_name, end_station_name

start_station_name  ... Average_time_for_trip
0                                       Zilker Park West  ...             22.636364
1                                Toomey Rd @ South Lamar  ...             48.985782
2                                       Waller & 6th St.  ...             38.334477
3                                             Pease Park  ...             30.497006
4                                       Waller & 6th St.  ...             18.060000
...                                                  ...  ...                   ...
13962  South Congress & Barton Springs at the Austin ...  ...             29.971429
13963  South Congress & Barton Springs at the Austin ...  ...             49.000000
13964  South Congress & Barton Springs at the Austin ...  ...             39.500000
13965  South Congress & Barton Springs at the Austin ...  ...             42.642857
13966  South Congress & Barton Springs at the Austin ...  ...             47.509804

[13967 rows x 3 columns]

There is 13967 rows in above dataset. That means there are that many number of  trips. Last column on right side of the table represents average time for every indiviual trip.